### DISCUSSION TITLE
Async tests

### DISCUSSION THREAD

**Discussion by mojixcoder** [collerek/ormar#408] [Ideas]
Is there any tests for async tests in FastAPI?

**Comment by collerek** [comment]
As of now - no, as the starlette test client is sync only, that would require usage of another package (like httpx).

If you need those it can be easily changed following the fastapi guide: https://fastapi.tiangolo.com/advanced/async-tests/

**Reply by mojixcoder** [reply]
Yeah tried that but I keep getting no running database error

**Comment by lewoudar** [comment]
I also have a small project using [async-asgi-testclient](https://github.com/vinissimus/async-asgi-testclient). If it can help you, can see it [here](https://github.com/lewoudar/tic-tac-toe/blob/main/tests/conftest.py) and [here](https://github.com/lewoudar/tic-tac-toe/blob/main/tests/test_update_game.py)

**Comment by mojixcoder** [comment]
@lewoudar That was awesome thank you.

**Reply by lewoudar** [reply]
You are welcome 🙂
